crates/pixi-build-python/src/metadata.rs

```diff
@@ -37,9 +38,24 @@ impl PyprojectMetadataProvider {
             manifest_root: manifest_root.into(),
             pyproject_manifest: OnceCell::default(),
             ignore_pyproject_manifest,
+            warnings: RefCell::new(Vec::new()),
         }
     }
 
+    /// Returns all warnings collected during metadata extraction.
+    ///
+    /// This includes warnings about invalid SPDX license expressions and other
+    /// metadata parsing issues that don't cause errors but may indicate problems.
+    #[allow(dead_code)]
```

Comment:
```suggestion
    #[expect(unused)]
```
This is nicer, since it fails when you start using it. So you can't forget to remove this attribute